megamek/src/megamek/client/bot/princess/FiringPlanCalculationParameters.java

```diff
@@ -0,0 +1,77 @@
+package megamek.client.bot.princess;
+
+import java.util.Map;
+
+import megamek.common.Entity;
+import megamek.common.Mounted;
+import megamek.common.Targetable;
+import megamek.common.annotations.Nullable;
+
+// This data structure contains parameters that may be passed to the "determineBestFiringPlan()"
+public class FiringPlanCalculationParameters
+{
+	//The type of firing plan calculation to carry out
+	public enum FiringPlanCalculationType
+	{
+		// We're guessing the firing plan based on our estimate of enemy movement
+		Guess,
+		// We're getting a firing plan based on exact known enemy movement results
+		Get
+	}
```

Comment:
FYI, enum values should be all caps by the Java coding standard as they're constants.